.github/workflows/release.yml

```diff
@@ -278,19 +278,55 @@ jobs:
           packages_dir: wheels/
           repository_url: https://test.pypi.org/legacy/
 
-  build-docs:
+  create-binder-env:
     needs: [deploy]
     runs-on: ubuntu-latest
+    steps:
+      - name: Get scikit-decide release version and define new binder env tag
+        run: |
+          TAG_NAME=${GITHUB_REF/refs\/tags\//}  # stripping "refs/tags/"
+          SKDECIDE_VERSION=${TAG_NAME/v/}  # stripping "v"
+          BINDER_TAG=binder_${TAG_NAME}
+          echo "BINDER_TAG=${BINDER_TAG}" >> $GITHUB_ENV
+          echo "SKDECIDE_VERSION=${SKDECIDE_VERSION}" >> $GITHUB_ENV
+
+      # Checks-out binder branch
+      - name: checkout binder branch
+        uses: actions/checkout@v2
+        with:
+          ref: binder
+
+      - name: Update scikit-decide dependency and tag the environement
+        run: |
+          sed -e "s/scikit-decide\[all\]==.*/scikit-decide\[all\]==${SKDECIDE_VERSION}/" environment.yml > environment.yml.new
+          mv environment.yml.new environment.yml
+          git config user.name "Binder Bot"
+          git config user.email "binder@bot.com"
+          git commit environment.yml -m "Update scikit-decide dependency"
+          git tag -a ${BINDER_TAG} -m "Release binder environment for scikit-decide ${SKDECIDE_VERSION}"
+          git push origin ${BINDER_TAG}
+
+      - name: Cache binder build on mybinder.org
+        uses: jupyterhub/repo2docker-action@master
+        with:
+          NO_PUSH: true
+          MYBINDERORG_TAG: ${{ env.BINDER_TAG }}
+
+  build-docs:
+    needs: [create-binder-env]
+    runs-on: ubuntu-latest
     env:
       DOCS_VERSION_PATH: /
 
     steps:
       - name: Set env variables for github+binder links in doc
         run: |
+          TAG_NAME=${GITHUB_REF/refs\/tags\//}  # stripping "refs/tags/"
```

Comment:
TAG_NAME could be put into GITHUB_ENV

Reply:
What for ? Maybe for DOCS_VERSION_PATH ? (i did not see that i could reuse something here also) But it should be SKDECIDE_VERSION in this case